DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,155 @@
+
+For example:
+```
+git checkout v0.13.8
+```
+
+Note that you can view the latest tags with:
+```
+git log --oneline --tags
+```
+
+You may also skip this tagged checkout and build the unreleased `master` branch instead.
+
+Another option is to "unpack" the latest release locally. This creates another directory containing the code for that release, but this code is not tracked by git, so it is less convenient if you are planning on making changes:
+
+```
+stack unpack purescript
+cd purescript-x.y.z  # (replace x.y.z to match the directory just created via unpack)
+```
+
+## Build
+
+```
+stack build
+```
+
+This will take a while the first time it is run.
+
+## Install
+
+If you'd like to test the build in a local shell, you can run:
+```
+stack exec bash
+```
+
+This prepends the location of the compiler to your `$PATH`, so you can test your changes. This works with other tools that depend on the `purs` binary, such as `spago`. Note that rebuilds continue to use this same path, so you don't need to keep re-running `stack exec` to test new changes (assuming you're using a separate terminal for `stack exec`). Remember that `stack exec` must be launched from the same directory that you ran `stack build` from.
+
+You may also install globally with:
+```
+stack install
+```
+
+This copies the compiler executable (`purs`) into `~/.local/bin`.
+
+## Profiling
+
+A profiling build is used to help diagnose performance issues with the compiler.
+
+Create a profiling build with:
+```
+stack build --profile
+```
+This will also take a while the first time it is run. Incremental builds are faster, and you may swap between standard and profiling builds without a rebuild penalty as work for each is saved in separate directories.
+
+Setting-up a local shell for your profiling build is similar to the steps for the standard build, just add the `--profile` flag:
+```
+stack exec --profile bash
+```
+Note that the bin directory prepended to `$PATH` is different than the standard build, so you can let this be a third "profiling" shell that you leave open between rebuilds.
+
+The `purs` compiler is often wrapped by `spago`. Here's how to pass the "time profiling" flag `-p` via spago:
+```
+spago build --purs-args "+RTS -p -RTS"
+```
+
+Note: There are other profiling flags (such as `-hc` for heap size). You can read more about these flags [here](http://book.realworldhaskell.org/read/profiling-and-optimization.html).
+
+This creates a `purs.prof` file. You can view the contents of this file directly, but it is often more convenient to use a visualizer.
+
+One nice `.prof` visualizer is [`profiteur`](https://github.com/jaspervdj/profiteur).
+```
+stack install profiteur
+profiteur purs.prof
+```
+Then open the generated `purs.prof.html` in your web browser.
+
+## Testing
+
+See the [Contributing](CONTRIBUTING.md) readme for testing instructions.
+
+## Haskell IDE
+
+There are many Haskell IDE options. Feel free to contribute instructions for your preferred editor.
```

Comment:
I don't really want to maintain instructions for particular IDE projects or editors here. There's nothing particularly weird about the PureScript compiler that should necessitate tooling or editor-specific guides, and if we're including instructions for HIE and VSCode, why should we not also include instructions for emacs, vim, ghcid, ghcide, etc.? I think this file will very quickly get out of hand (and probably also out of date) with this approach. This sort of documentation is best placed in the particular tooling/editor plugin projects themselves, I think.

Reply:
I'll move this IDE section to a Discourse post.